### PR TITLE
Control linux startup via advanced setting in gui

### PR DESCRIPTION
**Pull request by zanderz**
Should cover normal gnome case, didn't test kde

### CODE REVIEW

File: shared/desktop/app/app-state.js [hunk 1]
```diff
@@ -220,6 +235,51 @@ export default class AppState {
     app.setLoginItemSettings({openAtLogin: !!this.state.openAtLogin})
   }
 
+  setLinuxLoginState() {
+    const homeDir = String(process.env.HOME)
```
Comment (chrisnojima) [inline shared/desktop/app/app-state.js]: @oconnor663 can you look at this logic

**Comment by oconnor663** [comment]
I some worries, and I'm starting to think that solving these with the edit-file-in-place approach isn't worth the trouble, compared to just switching to a create-a-sentinel-file approach (below). Here are the worries:

- I haven't seen the `SESSIONTYPE` env var before. I have `XDG_SESSION_TYPE` on my system, but no `SESSIONTYPE`. Is there a consistent env var that we can rely on across different distros?
- Doing different things based on `isGnome` is going to run into tricky corner cases. A user might use multiple desktop environments on the same system. Maybe some crazy people switch DE's every day, but anyway doing a one-time switch is sort of normal. There are also environments where people network-sync the same home dir across multiple different machines, which might run different DE's. In general `isGnome` is a piece of runtime environment state, but anything we write to disk is going to outlive the environment, and it needs to make sense if the environment changes later. If we do need to set both `X-GNOME-Autostart-enabled` and `Hidden`, we should probably just set both all the time.
- Are we sure we need `X-GNOME-Autostart-enabled`? I run GNOME myself, and the `Hidden` setting by itself works fine for me.
- Why have a `\n` in only 1 of the 4 substitution strings?
- Using `sed` to edit the file in place is going to be very fragile. Right now we look for `#Hidden=true` exactly, so if the user puts a space in between the `#` and the `H`, that breaks us. (On random metadata files I wouldn't care about this, but here we've unfortunately told people to edit this file by hand.) Handling variable white space in all the places it might come up is nasty. We might also replace `#Hidden=true` if it wasn't the leftmost thing in its line.
- There are other Linux utilities like `gnome-tweak-tool`/`gnome-tweaks` that interact with `~/.config/autostart` files by just deleting them, rather than trying to edit them in place. The way we treat "missing" as "unconfigured" (preexisting problem, my fault!) runs afoul of those tools.

I'm starting to strongly prefer an alternative approach, which has the upside of hopefully being a lot less work for you guys:

- When core starts up, it can create both `~/.config/autostart/keybase-autostart.desktop` (as it does now) and also a sentinel file `~/.config/keybase/autostart-created`. As long as the sentinel exists, it will never try to recreate the former.
- Desktop can just ignore the sentinel, and only ever create/destroy the autostart file in response to an explicit user action. (We could put the text of the file under `/opt/keybase` if we want to be super duper DRY, or we could just duplicate it across core and desktop. Whatever.)
- We never need to edit any files in place, but user edits are still respected.

**Comment by chrisnojima** [comment]
k given this, lets just hide the checkbox @zanderz

**Comment by oconnor663** [comment]
I do think the sentinel approach would be even simpler than the code we have here. We'd want to do a core PR at the same time, to make sure we create the sentinel file before the UI starts potentially deleting the autostart file, but I'd be happy to write it.

**Comment by chrisnojima** [comment]
we're just not going to allow the checkbox to work on linux short term

**Pull request closed without merging**